### PR TITLE
feat: homogeneous Spektral support

### PR DESCRIPTION
**Pull request by parkererickson**
Support for homogeneous graphs and Spektral. Tested with `EdgeNeighborLoader`, `GraphLoader`, and `NeighborLoader`.

### CODE REVIEW

File: pyTigerGraph/gds/gds.py [hunk 1]
```diff
@@ -611,7 +611,7 @@ def graphLoader(
                 Defaults to None.
             output_format (str, optional):
                 Format of the output data of the loader.
-                Only "PyG", "DGL" and "dataframe" are supported. Defaults to "dataframe".
+                Only "PyG", "DGL", "spektral", and "dataframe" are supported. Defaults to "dataframe".
```
Comment (a-m-thomas) [inline pyTigerGraph/gds/gds.py:614]: No need to say "only" here if it's not in neightborLoader or edgeNeighborLoader

**Review by a-m-thomas — APPROVED** [review]

File: pyTigerGraph/gds/dataloaders.py [hunk 2]
```diff
@@ -936,71 +993,65 @@ def add_sep_attr(attr_names: list, attr_types: dict, attr_df: pd.DataFrame,
...
                             edges[etype] = subdf1
                             edgelist[etype] = edges[etype][["tmp_id_x", "tmp_id_y"]]
-                else:
-                    for etype in edges:
-                        edgelist[etype] = edges[etype][["source", "target"]]
+            else:
                 for etype in edges:
-                    edgelist[etype] = torch.tensor(edgelist[etype].to_numpy().T, dtype=torch.long)
-                if mode == "dgl":
-                    data = dgl.heterograph({
```
Comment (tg-bill) [inline pyTigerGraph/gds/dataloaders.py:945]: the DGL heterograph commit is overwritten

**Review by tg-bill — COMMENTED** [review]
Some recent commits are not included in this PR. It will be a really tedious task to get the merge right.

**Comment by parkererickson** [comment]
I think I got the recent stuff merged in now.

**Review by tg-bill — APPROVED** [review]

**Pull request merged**